src/query/procedure/py_module.cpp

```diff
@@ -905,30 +918,66 @@ std::optional<py::ExceptionInfo> AddRecordFromPython(mgp_result *result, py::Obj
       PyErr_SetString(PyExc_TypeError, msg.c_str());
       return py::FetchError();
     }
-    const auto *field_name = PyUnicode_AsUTF8(key);
+    const char *field_name = PyUnicode_AsUTF8(key);
     if (!field_name) return py::FetchError();
-    auto *val = PyTuple_GetItem(item, 1);
+    PyObject *val = PyTuple_GetItem(item, 1);
     if (!val) return py::FetchError();
     // This memory is one dedicated for mg_procedure.
     mgp_value *field_val = PyObjectToMgpValueWithPythonExceptions(val, memory);
     if (field_val == nullptr) {
       return py::FetchError();
     }
-    if (mgp_result_record_insert(record, field_name, field_val) != mgp_error::MGP_ERROR_NO_ERROR) {
+
+    if (!is_transactional) {
+      if (ContainsDeleted(field_val)) {
+        mgp_value_destroy(field_val);
+        for (auto &cache_entry : current_record_cache) {
+          mgp_value_destroy(cache_entry.field_val);
+        }
+
+        return std::nullopt;
```

Comment:
I would make a comment here that if one is deleted, we are removing allo of them from record, because whole record gets scraped

Reply:
Done